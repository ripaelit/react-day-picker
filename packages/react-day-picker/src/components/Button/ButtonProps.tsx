import * as React from 'react';

/** The props for the [[Button]] component. */

export type ButtonProps = React.HTMLProps<HTMLButtonElement>;
