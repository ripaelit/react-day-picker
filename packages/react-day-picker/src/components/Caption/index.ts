export * from './Caption';
export * from './getCaptionProps';
export * from './types';
