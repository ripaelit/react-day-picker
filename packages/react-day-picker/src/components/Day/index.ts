export * from './Day';
export * from './utils/getModifiersStatus';
