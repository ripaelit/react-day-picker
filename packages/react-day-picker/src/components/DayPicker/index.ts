export * from './DayPicker';
// export * from './DayPickerContext';
// export * from './NavigationContext';
