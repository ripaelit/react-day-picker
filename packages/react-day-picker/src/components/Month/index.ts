export * from './Month';
export * from './types';
