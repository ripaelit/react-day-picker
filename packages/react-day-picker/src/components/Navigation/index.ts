export * from './getNavigationProps';
export * from './Navigation';
export * from './types';
export * from './utils';
