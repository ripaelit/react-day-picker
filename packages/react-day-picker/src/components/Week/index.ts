export { Week as WeekRow } from './Week';
export * from './types';
