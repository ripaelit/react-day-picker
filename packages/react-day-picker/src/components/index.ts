export * from './Caption';
export * from './Day';
export * from './DayPicker';
export * from './Dropdown';
export * from './Head';
export * from './IconDropdown';
export * from './IconNext';
export * from './IconPrev';
export * from './MonthsDropdown';
export * from './Root';
export * from './Row';
export * from './Table';
export * from './WeekNumber';
export * from './YearsDropdown';
