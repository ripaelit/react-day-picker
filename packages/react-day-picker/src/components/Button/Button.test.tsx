import { RenderResult } from '@testing-library/react';
import React from 'react';

import { customRender } from 'test';

import { Button } from 'components';

import { ButtonProps } from './ButtonProps';

const setup = (
  props?: ButtonProps & React.RefAttributes<HTMLButtonElement>
): RenderResult => {
  return customRender(<Button {...props} />);
};
test('should render correctly', () => {
  const { container } = setup();
  expect(container.firstChild).toMatchSnapshot();
});

test('should add the class name from props', () => {
  const { container } = setup({ className: 'foo' });
  expect(container.firstChild).toMatchSnapshot();
});

test('should add the inline style from props', () => {
  const { container } = setup({ style: { color: 'red' } });
  expect(container.firstChild).toMatchSnapshot();
});
