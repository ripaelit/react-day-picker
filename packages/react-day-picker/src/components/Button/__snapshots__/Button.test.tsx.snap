// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add the class name from props 1`] = `
<button
  class="rdp-button_reset rdp-button foo"
  type="button"
/>
`;

exports[`should add the inline style from props 1`] = `
<button
  class="rdp-button_reset rdp-button"
  style="color: red;"
  type="button"
/>
`;

exports[`should render correctly 1`] = `
<button
  class="rdp-button_reset rdp-button"
  type="button"
/>
`;
